### PR TITLE
[TEVA-2070] refactor notification component

### PR DESCRIPTION
**Pull request by alexbowen**
https://dfedigital.atlassian.net/browse/TEVA-2070 (one component of)

- aligns component with agreed consistent approach
- uses already implemented generic component rspec tests
- allows us to use the most recent view components gem
- also makes component work nicely as a preview component also that will be in subsequent PR

### CODE REVIEW

**Comment by martin-bangoura** [comment]
Review app deployed to <https://teaching-vacancies-review-pr-3084.london.cloudapps.digital>

**Comment by alexbowen** [comment]
Metrics/ParameterLists: Avoid parameter lists longer than 8 parameters - what would be the best way to solve this rubocop error? obviously i know what it means - should i convert some arguments into a hash?

File: app/components/shared/notification_component.rb [hunk 1]
```diff
@@ -1,36 +1,61 @@
...
+    if @variant == "empty"
       {}
     else
       { role: "alert", tabindex: "-1" }
     end
   end
+
+  private
+
+  def default_classes
```
Comment (csutter) [inline app/components/shared/notification_component.rb:53]: Slightly tidier maybe:

```ruby
["notification-component", "govuk-notification", variant_class].tap do |applied_classes|
  applied_classes.push(background_class) if @background
  applied_classes.push(icon_class) if @icon
  applied_classes.push(dismissable_class) if @dismiss
end
```

**Review by csutter — APPROVED** [review]
Some minor Ruby nitpicking but looks great otherwise 👍

File: app/components/shared/notification_component.rb [hunk 2]
```diff
@@ -1,36 +1,61 @@
-class Shared::NotificationComponent < ViewComponent::Base
-  def initialize(content:, style:,
+class Shared::NotificationComponent < GovukComponent::Base
+  attr_accessor :variant, :links, :title, :body, :dismiss, :background, :icon
+
+  # rubocop:disable Metrics/ParameterLists
+  def initialize(variant: "success",
```
[the remark this answers is not in the document]
Comment (alexbowen) [reply app/components/shared/notification_component.rb:5]: will do this seperatly if that ok

**Pull request merged**